### PR TITLE
update SSH checks with full list of FIPS Ciphers and MACs

### PR DESCRIPTION
**Pull request by shawndwells**
from cipher+MACs in the FIPS paperwork at:

http://csrc.nist.gov/groups/STM/cmvp/documents/140-1/140sp/140sp2630.pdf

### CODE REVIEW

File: shared/oval/sshd_use_approved_macs.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
   </ind:textfilecontent54_test>
   <ind:textfilecontent54_object id="obj_sshd_use_approved_macs" version="1">
     <ind:filepath>/etc/ssh/sshd_config</ind:filepath>
-    <ind:pattern operation="pattern match">^[\s]*(?i)MACs(?-i)[\s]+hmac-sha2-512,hmac-sha2-256,hmac-sha1[\s]*(?:|(?:#.*))?$</ind:pattern>
+    <ind:pattern operation="pattern match">^[\s]*(?i)MACs(?-i)[\s]+((hmac-sha1|hmac-sha2-256|hmac-sha2-512|hmac-sha1-etm@openssh.com|hmac-sha2-256-etm@openssh.com|hmac-sha2-512-etm@openssh.com),?)+[\s]*(?:|(?:#.*))?$</ind:pattern>
```
Comment (dahaic) [inline shared/oval/sshd_use_approved_macs.xml]: I think those "." should be escaped. (In ciphers as well)

**Comment by shawndwells** [comment]
On 4/28/17 12:51 PM, Marek Haičman wrote:
>
> *@dahaic* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In shared/oval/sshd_use_approved_macs.xml
> <https://github.com/OpenSCAP/scap-security-guide/pull/1976#discussion_r113973516>:
>
> > @@ -25,7 +25,7 @@
>    </ind:textfilecontent54_test>
>    <ind:textfilecontent54_object id="obj_sshd_use_approved_macs" version="1">
>      <ind:filepath>/etc/ssh/sshd_config</ind:filepath>
> -    <ind:pattern operation="pattern match">^[\s]*(?i)MACs(?-i)[\s]+hmac-sha2-512,hmac-sha2-256,hmac-sha1[\s]*(?:|(?:#.*))?$</ind:pattern>
> +    <ind:pattern operation="pattern match">^[\s]*(?i)MACs(?-i)[\s]+((hmac-sha1|hmac-sha2-256|hmac-sha2-512|hmac-sha1-etm@openssh.com|hmac-sha2-256-etm@openssh.com|hmac-sha2-512-etm@openssh.com),?)+[\s]*(?:|(?:#.*))?$</ind:pattern>
>
> I think those "." should be escaped. (In ciphers as well)
>

You're totally right. Updated. Thanks!

File: shared/oval/sshd_use_approved_ciphers.xml [hunk 2]
```diff
@@ -26,7 +26,7 @@
   </ind:textfilecontent54_test>
   <ind:textfilecontent54_object id="obj_sshd_use_approved_ciphers" version="2">
     <ind:filepath>/etc/ssh/sshd_config</ind:filepath>
-    <ind:pattern operation="pattern match">^[\s]*(?i)Ciphers(?-i)[\s]+((aes128-ctr|aes192-ctr|aes256-ctr|aes128-cbc|3des-cbc|aes192-cbc|aes256-cbc),?)+[\s]*(?:|(?:#.*))?$</ind:pattern>
+    <ind:pattern operation="pattern match">^[\s]*(?i)Ciphers(?-i)[\s]+((aes128-ctr|aes192-ctr|aes256-ctr|aes128-cbc|aes192-cbc|aes256-cbc|3des-cbc|rinjndael-cbc@lysator\.liu\.se),?)+[\s]*(?:|(?:#.*))?$</ind:pattern>
```
Comment (dahaic) [inline shared/oval/sshd_use_approved_ciphers.xml]: Sorry, one more typo. It should be ```rijndael-cbc```. The first "n" shouldn't be there.

Comment (dahaic) [inline shared/oval/sshd_use_approved_ciphers.xml]: Same typo is in the text

**Comment by shawndwells** [comment]
On 4/28/17 1:01 PM, Marek Haičman wrote:
>
> *@dahaic* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In shared/oval/sshd_use_approved_ciphers.xml
> <https://github.com/OpenSCAP/scap-security-guide/pull/1976#discussion_r113975448>:
>
> > @@ -26,7 +26,7 @@
>    </ind:textfilecontent54_test>
>    <ind:textfilecontent54_object id="obj_sshd_use_approved_ciphers" version="2">
>      <ind:filepath>/etc/ssh/sshd_config</ind:filepath>
> -    <ind:pattern operation="pattern match">^[\s]*(?i)Ciphers(?-i)[\s]+((aes128-ctr|aes192-ctr|aes256-ctr|aes128-cbc|3des-cbc|aes192-cbc|aes256-cbc),?)+[\s]*(?:|(?:#.*))?$</ind:pattern>
> +    <ind:pattern operation="pattern match">^[\s]*(?i)Ciphers(?-i)[\s]+((aes128-ctr|aes192-ctr|aes256-ctr|aes128-cbc|aes192-cbc|aes256-cbc|3des-cbc|rinjndael-cbc@lysator\.liu\.se),?)+[\s]*(?:|(?:#.*))?$</ind:pattern>
>
> Sorry, one more typo. It should be |rijndael-cbc|. The first "n"
> shouldn't be there.
>

@dahaic - updated

**Comment by mpreisler** [comment]
Bad XML https://jenkins.open-scap.org/job/scap-security-guide-pull-requests/label=node-el6-openscap-new/1449/console

**Comment by mpreisler** [comment]
Might be a problem with jenkins

@openscap-jenkins test this please

**Comment by shawndwells** [comment]
builds locally, so guessing jenkins:

`````
$ cmake ../ ; make -j4 rhel7
.....
[100%] Built target generate-ssg-rhel7-role.yml
Generated '/var/www/html/scap-security-guide/build/roles/ssg-rhel7-role-nist-800-171-cui.sh' for profile ID 'xccdf_org.ssgproject.content_profile_nist-800-171-cui' in benchmark 'xccdf_org.ssgproject.content_benchmark_RHEL-7', template=urn:xccdf:fix:script:sh.
[100%] Built target generate-ssg-rhel7-role.sh
Scanning dependencies of target rhel7-roles
[100%] Built target rhel7-roles
[100%] Built target rhel7
`````

**Comment by shawndwells** [comment]
@mpreisler - jenkins passes now

**Comment by mpreisler** [comment]
ACK

**Pull request merged**